pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -179,7 +179,8 @@ func (m *serviceAffinityMetadata) clone() *serviceAffinityMetadata {
 	return &copy
 }
 
-type podAffinityMetadata struct {
+// PodAffinityMetadata pre-computed state for inter-pod affinity predicate.
+type PodAffinityMetadata struct {
```

Comment:
We need to export this and other related functions so that we can generate it via the PreFilter